From 3a93f4dc7d0d2ccb0d8339080b7638c3e8b999d7 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Wed, 8 Jan 2025 12:12:15 +0100
Subject: [PATCH] fmf: Only mess with /etc/yum.repos.d when running within
 testing farm

If running tmt locally to debug the test script, make sure we don't
mess with /etc/yum.repos.d.

(cherry picked from commit 8e3347f3bd3d9a01b8f39b0858eab74084ecf20a)
---
 test/fmf/integration-tests/test.sh | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/test/fmf/integration-tests/test.sh b/test/fmf/integration-tests/test.sh
index 0a1595fa97..347cd219a4 100755
--- a/test/fmf/integration-tests/test.sh
+++ b/test/fmf/integration-tests/test.sh
@@ -63,6 +63,11 @@ Release=${VERSION_ID:-rawhide}
 [Build]
 ToolsTreeDistribution=$ID
 ToolsTreeRelease=${VERSION_ID:-rawhide}
+EOF
+
+if [[ -n "${TESTING_FARM_REQUEST_ID:-}" ]]; then
+    tee --append mkosi.local.conf <<EOF
+[Build]
 ToolsTreeSandboxTrees=
         /etc/yum.repos.d/:/etc/yum.repos.d/
         /var/share/test-artifacts/:/var/share/test-artifacts/
@@ -72,14 +77,15 @@ SandboxTrees=
 Environment=NO_BUILD=1
 EOF
 
-cat /etc/dnf/dnf.conf
-cat /etc/yum.repos.d/*
+    cat /etc/dnf/dnf.conf
+    cat /etc/yum.repos.d/*
 
-# Ensure packages built for this test have highest priority
-echo -e "\npriority=1" >> /etc/yum.repos.d/copr_build*
+    # Ensure packages built for this test have highest priority
+    echo -e "\npriority=1" >> /etc/yum.repos.d/copr_build*
 
-# Disable mkosi's own repository logic
-touch /etc/yum.repos.d/mkosi.repo
+    # Disable mkosi's own repository logic
+    touch /etc/yum.repos.d/mkosi.repo
+fi
 
 # TODO: drop once BTRFS regression is fixed in kernel 6.13
 sed -i "s/Format=btrfs/Format=ext4/" mkosi.repart/10-root.conf
